messaging/src/main/java/org/axonframework/messaging/MessageTypeResolver.java

```diff
@@ -37,12 +39,13 @@ public interface MessageTypeResolver {
      *
      * @param payload The {@link Message#getPayload() Message payload} to resolve a {@link MessageType type} for.
      * @return The {@link MessageType type} for the given {@code payload}.
+     * @throws MessageTypeNotResolvedException if the {@link MessageType type} could not be resolved.
      */
-    default MessageType resolve(Object payload) {
+    default MessageType resolveOrThrow(Object payload) {
```

Comment:
I really love how explicit this method is. Really good! :)

Reply:
I know - my Kotlin buddy :D